src/display/editor/highlight.js

```diff
@@ -693,15 +693,15 @@ class HighlightEditor extends AnnotationEditor {
     let i = 0;
     for (const { x, y, width, height } of boxes) {
       const sx = x * pageWidth + pageX;
-      const sy = (1 - y - height) * pageHeight + pageY;
+      const sy = (1 - y) * pageHeight + pageY;
       // The specifications say that the rectangle should start from the bottom
       // left corner and go counter-clockwise.
       // But when opening the file in Adobe Acrobat it appears that this isn't
       // correct hence the 4th and 6th numbers are just swapped.
       quadPoints[i] = quadPoints[i + 4] = sx;
       quadPoints[i + 1] = quadPoints[i + 3] = sy;
       quadPoints[i + 2] = quadPoints[i + 6] = sx + width * pageWidth;
-      quadPoints[i + 5] = quadPoints[i + 7] = sy + height * pageHeight;
+      quadPoints[i + 5] = quadPoints[i + 7] = sy - height * pageHeight;
```

Comment:
If I understand correctly (b=bottom, t=top, L=left, R=right):
 - before your patch: the order is bL, bR, tL, tR;
 - after your patch: the order is tL, tR, bL, bR.

I just tested in Acrobat (I highlighted a word, saved and checked the content of the pdf) and the saved order is bL, bR, tL, tR.

I proposed you just change the order here to respect the specification, I mean having bL, bR, tR, tL, this way we don't change the way to deserialize.
But as I said it'd require to update https://github.com/mozilla/pdf.js/blob/3219d4905df61516e7097fdc7211de569d90584e/src/core/annotation.js#L4686 to save the quadPoints in the same way as Acrobat.

Usually, we try to follow the official pdf specs but from time to time we follow what's Acrobat does.
Here what's you're proposing does conform to your own expectations but neither to the specs nor to Acrobat.

Reply:
@calixteman I just checked the acrobat reader and tried to save a highlight. I use a 3rd library from python to read the quadPoints from that file and found that the correct order after saved should be tL, tR, bL, bR (note: the PDF coordinate is bottom left, then the greater y value mean higher position)

<img width="1218" alt="image" src="https://github.com/user-attachments/assets/4b21f4a9-7788-46d5-b073-b197f923ac2e">

Reply:
You're right, the order in Acrobat is tL, tR, bL, bR (I forgot the origin of the page is in the bottom-left).